liota/lib/transports/amqp.py

```diff
@@ -0,0 +1,343 @@
+        :type routing_key: str or unicode
+        :param message: Message to be published
+        :type message: str or unicode
+        :param bool exchange_durable: Value to survive exchange on RabbitMQ reboot
+        :return:
+        """
+        # Declare exchange
+        self.channel.exchange_declare(exchange=exchange_name,
+                                      type='topic',
+                                      durable=exchange_durable)
+
+        # Publish acknowledgement
+        publish_ack = self.channel.basic_publish(exchange=exchange_name,
+                                                 routing_key=routing_key,
+                                                 body=message)
+
+        # Checking message delivery confirmation
+        if not publish_ack:
+            log.error("AMQP Publish error for routing key: " + routing_key)
+            raise ValueError("AMQP publish error for routing key: " + routing_key)
+        else:
+            log.info("AMQP message published successfully")
+
+    def consume(self, exchange_name, routing_key, exchange_durable, queue_durable, no_ack=False, callback=None):
+        """
+        :param exchange_name: Exchange name
+        :type exchange_name: str or unicode
+        :param routing_key: Routing key for binding
+        :type routing_key: str or unicode
+        :param bool exchange_durable: Value to survive exchange on RabbitMQ reboot
+        :param bool queue_durable: Value to survive queue on RabbitMQ reboot
+        :param bool no_ack: No acknowledgement value for consumer
+        :param callable callback: function for dispatching messages to user having the signature:
+                                  consumer_callback(channel, method, properties, body)
+        :return:
+        """
+        # Set QoS parameters
+        if self.qos_details:
+            self.channel.basic_qos(self.qos_details.prefetch_size, self.qos_details.prefetch_count,
+                                   self.qos_details.all_channels)
+
+        # Declare exchange
+        self.channel.exchange_declare(exchange=exchange_name,
+                                      type='topic',
+                                      durable=exchange_durable)
+
+        # Declare queue
+        result = self.channel.queue_declare(durable=queue_durable)
+
+        # Extract queue name
+        queue_name = result.method.queue
+
+        # Bind queue with exchange and routing key
+        self.channel.queue_bind(exchange=exchange_name,
+                                queue=queue_name,
+                                routing_key=routing_key)
+
+        # Start message consuming, providing consumer callback
+        self.channel.basic_consume(consumer_callback=callback,
+                                   queue=queue_name,
+                                   no_ack=no_ack)
+        try:
+            self.channel.start_consuming()
+        except Exception, e:
+            raise e
```

Comment:
`publish` and `consume` should not raise exception. use `log.exception` instead.  However, connect and disconnect should raise exception.